src/lib/moment/locale.js

```diff
@@ -1,33 +1,31 @@
+import { Moment } from './constructor';
 import { getLocale } from '../locale/locales';
-import { deprecate } from '../utils/deprecate';
 
-// If passed a locale key, it will set the locale for this
-// instance.  Otherwise, it will return the locale configuration
-// variables for this instance.
+// If passed a locale key, it will return a cloned instance that is set
+// to the specified locale.  Otherwise, it will return the name of the
+// locale that is set on this instance.
 export function locale (key) {
-    var newLocaleData;
+    var clone, newLocaleData;
 
     if (key === undefined) {
         return this._locale._abbr;
     } else {
+        clone = new Moment(this);
         newLocaleData = getLocale(key);
         if (newLocaleData != null) {
-            this._locale = newLocaleData;
+            clone._locale = newLocaleData;
```

Comment:
I get a bit uneasy when reading this. Sure, its a new clone, but I can see a bug slipping through if one is not very careful. Would it be possible to cram all the "mutability" inside the constructor, so it gets all arguments and produces a frozen (maybe not literally) moment.